sbi/simulators/gaussian_mixture.py

```diff
@@ -0,0 +1,121 @@
+import logging
```

Comment:
I would consider splitting this PR with a first PR for this GMM simulator.

Reply:
it would also force you to add a test for the simulator :)

Reply:
Good Idea, but I won't have the time. @michaeldeistler , @janfb what do you think?

Reply:
Do we really need the Gaussian mixture simulator for the test? If so, I would add it analogously to the `linear_gaussian` simulator, i.e., just add the simulator as plain function and then add another function that can samples the reference posterior. The prior will be defined in the test then.

Also, the class looks very similar to the sbibm implementation. If we use that here we should add a note.

Reply:
I used the `gaussian_mixture` in the tutorial as an example and it was therefore easier for me to use it in the tests again. I also think it's a useful example more generally speaking for future tutorials. So I will keep it and make the changes you requested @janfb. 

If you want I can add a test with the `gaussian_linear` simulator too. Let me know!